src/main/java/org/cloudburstmc/server/entity/impl/BaseEntity.java

```diff
@@ -1285,7 +1286,9 @@ public AxisAlignedBB getBoundingBox() {
     public void fall(float fallDistance) {
         float damage = (float) Math.floor(fallDistance - 3 - (this.hasEffect(Effect.JUMP) ? this.getEffect(Effect.JUMP).getAmplifier() + 1 : 0));
         if (damage > 0) {
-            this.attack(new EntityDamageEvent(this, EntityDamageEvent.DamageCause.FALL, damage));
+            if (!this.isPlayer || level.getGameRules().get(GameRules.FALL_DAMAGE)) {
```

Comment:
If fall damage is disabled, wouldn't it make sense to quite before we even make the calculation right at the top of the method?